files/setup_freeipa.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# This script is called by the freeipa-enroll service.
+
+# These variables must be set before execution.  They will
+# be loaded from a file installed by cloud-init:
+# /var/lib/cloud/instance/freeipa-creds.sh
+# ADMIN_PW: The password for the IPA server's Kerberos admin role
+# HOSTNAME: The hostname of this IPA client (e.g. client.example.com)
+# REALM: The realm for the IPA server (e.g. EXAMPLE.COM)
+
+# shellcheck disable=SC1091
+source /var/lib/cloud/instance/freeipa-creds.sh
+
+# Get the default Ethernet interface
+function get_interface {
+    ip route | grep default | sed "s/^.* dev \([^ ]*\).*$/\1/"
+}
+
+# Get the IP address corresponding to an interface
+function get_ip {
+    ip --family inet address show dev "$1" | \
+        grep inet | \
+        sed "s/^ *//" | \
+        cut --delimiter=' ' --fields=2 | \
+        cut --delimiter='/' --fields=1
+}
+
+# Get the PTR record corresponding to an IP
+function get_ptr {
+    dig +noall +ans -x "$1" | sed "s/.*PTR[[:space:]]*\(.*\)/\1/"
+}
+
+function enroll {
+    # Check to see if freeipa-client is already installed
+    if [[ -f "/etc/ipa/default.conf" ]]; then
+        echo "FreeIPA client is already installed... rejoining."
+        echo "${ADMIN_PW}" | kinit admin@"${REALM}"
+        ipa-join || true
+        kdestroy
+        exit 0
+    fi
+
+    interface=$(get_interface)
+    ip_address=$(get_ip "$interface")
+
+    # Wait until the IP address has a non-Amazon PTR record before
+    # proceeding
+    ptr=$(get_ptr "$ip_address")
+    while grep amazon <<< "$ptr"
```

Comment:
Not sure if it's worth the effort, but maybe this should be more generic (i.e. for non-AWS users)?

Reply:
See commit ea1d82d

Reply:
👍